dev/validate_bom__test.py

```diff
@@ -593,6 +632,44 @@ def wait_on_service(self, service_name, port=None, timeout=None):
         ResponseError('It appears that {0} failed'.format(service_name),
                       server='tunnel'))
 
+  def __wait_on_service_using_host(self, service_name, timeout=None):
+    service_config = self.__outbound_service_configs[service_name]
+    host = service_config['host']
+
+    timeout = timeout or self.options.test_service_startup_timeout
+    end_time = time.time() + timeout
+    logging.info('Waiting on "%s..."', service_name)
+
+    threadid = threading.current_thread().ident
+    logging.info('WaitOn polling %s from thread %x', service_name, threadid)
+    try:
+      if 'bearer_auth_token' in service_config:
+        _make_get_request_with_bearer_auth_token('{host}/health'
+                                            .format(host=host),
+                                            service_config['bearer_auth_token'])
+      # Add more authentication cases here.
+      else:
+        urlopen('{host}/health'
+                .format(host=host))
+      logging.info('"%s" is ready on service host %s | %s',
```

Comment:
you used %x as a thread id before if you are trying to align them.
rather than " | %s " you could "threadid=%x" here and " from thread %x" with " threadid=%x" above.